### PR TITLE
Improvements for 32-bit systems

### PR DESCRIPTION
**Pull request by nwellnhof**

### CODE REVIEW

File: sort.h [hunk 1]
```diff
@@ -877,20 +877,21 @@ static void TIM_SORT_MERGE(SORT_TYPE *dst, const TIM_SORT_RUN_T *stack, const in
...
+    j = curr + A;
+    k = curr + A + B;
 
-    for (k = curr + A + B - 1; k >= curr; k--) {
-      if ((i >= 0) && (j >= curr)) {
-        if (SORT_CMP(dst[j], storage[i]) > 0) {
-          dst[k] = dst[j--];
+    while (k-- > curr) {
+      if ((i > 0) && (j > curr)) {
+        if (SORT_CMP(dst[j-1], storage[i-1]) > 0) {
```
Comment (swenson) [inline sort.h]: Could you run `make format` on this by any chance? I like to keep the style consistent.

File: sort.h [hunk 2]
```diff
@@ -877,20 +877,21 @@ static void TIM_SORT_MERGE(SORT_TYPE *dst, const TIM_SORT_RUN_T *stack, const in
...
-    j = curr + A - 1;
+    i = B;
+    j = curr + A;
+    k = curr + A + B;
 
-    for (k = curr + A + B - 1; k >= curr; k--) {
-      if ((i >= 0) && (j >= curr)) {
-        if (SORT_CMP(dst[j], storage[i]) > 0) {
-          dst[k] = dst[j--];
+    while (k-- > curr) {
```
Comment (swenson) [inline sort.h]: I'm curious why we switched this loop from being indexed by `i` and `j` to `i - 1` and `j - 1`. Is there a performance impact?

Comment (nwellnhof) [reply sort.h]: With unsigned indices, the condition like `i >= 0` is always true. The same with `j >= curr` and `k >= curr` if `curr` is zero.

File: sort.h (hunk 1 above)
Comment (nwellnhof) [reply sort.h]: Will do.

**Comment by nwellnhof** [comment]
Here's a branch with correct formatting and another minor optimization:

https://github.com/nwellnhof/sort/commits/optimize-timsort-merge

Here's a different approach that works directly on pointers when traversing the runs.

https://github.com/nwellnhof/sort/commit/a5245cc0dae89d2e8a5f49100f15622572bbb884

With simple sort functions, I get another 15% speed up on the register-starved x86-32 architecture. It makes the code harder to understand, though. Let me know which approach you prefer.

**Pull request closed without merging**